packages/maplibre/lib/map/apply-context-diff.ts

```diff
@@ -0,0 +1,61 @@
+import { MapContextDiff } from "@geospatial-sdk/core";
+import { Map } from "maplibre-gl";
+import { createLayer } from "./create-map";
+import { removeLayersFromSource } from "../helpers/map.helpers";
+
+/**
+ * Apply a context diff to an MapLibre map
+ * @param map
+ * @param contextDiff
+ */
+export async function applyContextDiffToMap(
+  map: Map,
+  contextDiff: MapContextDiff,
+): Promise<void> {
+  // removed layers (sorted by descending position)
+  if (contextDiff.layersRemoved.length > 0) {
+    const removed = contextDiff.layersRemoved.sort(
+      (a, b) => b.position - a.position,
+    );
+    for (const layerRemoved of removed) {
+      const maplibreLayer = map.getStyle().layers[layerRemoved.position];
+      if (maplibreLayer.type === "background") {
+        console.warn(
+          "[Warning] applyContextDiffToMap: removing background layer is not supported.",
+        );
+        continue;
+      }
+      const sourceId = maplibreLayer.source;
+      map.removeLayer(maplibreLayer.id);
+      map.removeSource(sourceId);
+    }
+  }
+
+  // insert added layers
+  const newLayers = await Promise.all(
+    contextDiff.layersAdded.map((layerAdded) => createLayer(layerAdded.layer)),
+  );
+  newLayers.forEach((style, index) => {
+    const position = contextDiff.layersAdded[index].position;
+    const beforeId = map.getLayersOrder()[position]; // can be undefined
```

Comment:
I think this is also assuming that 1 context layer => 1 maplibre layer, but this will actually not work in practice I think? anyway it can stay as a known limitation for now